### DISCUSSION TITLE
Type hinting for default built-in functions & signals

### DISCUSSION THREAD

**Discussion by XCarCedo** [godotengine/godot-proposals#7465] [Scripting]
As far as i know type hinting is a good thing to do to prevent errors and get better completion across your code, a function like _physics_process takes a delta time which is float or for example a body_entered function which will always give a PhysicsBody2D or TileMap, i'm not sure if this is possible to implement automatic type hinting in every area without being buggy so lmk if i'm wrong

**Comment by Calinou** [comment]
There's already an editor setting to add type hints when creating scripts from templates, or functions from signal connections in the script editor: **Text Editor > Completion > Type Hints**

**Comment by XCarCedo** [comment]
Oh okay, didn't know that thanks, and i don't thing type hints for things like a simple delta would be much of help 😅 so i'm closing the discussion
